### PR TITLE
Mrd 1974 no ppud record found appinsights

### PR DESCRIPTION
**Pull request by bills7281**

### CODE REVIEW

File: server/views/pages/recommendations/searchPpud.njk [hunk 1]
```diff
@@ -23,31 +23,39 @@
                         Name
                     </dt>
                     <dd class="govuk-summary-list__value">
+                    {% if recommendation.personOnProbation.fullName %}
                           {{ recommendation.personOnProbation.fullName }}
+                    {% else %}-{% endif %}
```
Comment (DarrenW-HMPPS) [inline server/views/pages/recommendations/searchPpud.njk:28]: Should these hyphens include the screen reader text saying "no value" or something similar?

**Review by DarrenW-HMPPS — CHANGES_REQUESTED** [review]

File: server/views/pages/recommendations/searchPpud.njk (hunk 1 above)
Comment (mayank-s18) [reply server/views/pages/recommendations/searchPpud.njk:28]: yes, hyphens should include the screen reader text saying "no value"

Comment (bills7281) [reply server/views/pages/recommendations/searchPpud.njk]: Done

File: server/views/pages/recommendations/searchPpud.njk [hunk 2]
```diff
@@ -23,31 +23,31 @@
...
+                    {{ hyphenForBlank(recommendation.personOnProbation.fullName) }}
                     </dd>
                 </div>
                 <div class="govuk-summary-list__row">
                     <dt class="govuk-summary-list__key">
                         Date of birth
                     </dt>
                     <dd class="govuk-summary-list__value">
-                        {{ formatDateTimeFromIsoString({isoDate: recommendation.personOnProbation.dateOfBirth}) }}
+                    {{ hyphenForBlank(recommendation.personOnProbation.dateOfBirth) }}
```
Comment (DarrenW-HMPPS) [inline server/views/pages/recommendations/searchPpud.njk:34]: Does this still format the date properly?

Comment (bills7281) [reply server/views/pages/recommendations/searchPpud.njk]: Thanks Darren, nicely spotted!

File: server/views/pages/recommendations/searchPpud.njk [hunk 3]
```diff
@@ -23,39 +23,31 @@
...
                 </div>
                 <div class="govuk-summary-list__row">
                     <dt class="govuk-summary-list__key">
                         Date of birth
                     </dt>
                     <dd class="govuk-summary-list__value">
-                    {% if recommendation.personOnProbation.dateOfBirth %}
-                        {{ formatDateTimeFromIsoString({isoDate: recommendation.personOnProbation.dateOfBirth}) }}
-                    {% else %}-{% endif %}
+                    {{ hyphenForBlank(formatDateTimeFromIsoString({isoDate: recommendation.personOnProbation.dateOfBirth})) }}
```
[the remark this answers is not in the document]
Comment (bills7281) [reply server/views/pages/recommendations/searchPpud.njk:41]: I tested it with both a null and missing date of birth and it appears to be behaving correctly.
I've attached a short video here.

https://github.com/ministryofjustice/make-recall-decision-ui/assets/104494371/448a82fe-2c7d-486d-8967-ba5083dff35a

[the remark this answers is not in the document]
Comment (bills7281) [reply server/views/pages/recommendations/searchPpud.njk:41]: I've tested it with both a null and a missing date of birth. It appears to be behaving itself.
Here is a screen capture showing it ll working ~

https://github.com/ministryofjustice/make-recall-decision-ui/assets/104494371/253225d7-bc05-4468-803b-68cb7e731933

**Pull request merged**